File: sparkJava/src/main/java/com/beifeng/sql/SaveModeTest.java
package com.beifeng.sql;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.sql.DataFrame;
import org.apache.spark.sql.SQLContext;
import org.apache.spark.sql.SaveMode;

/**
 * Created by youzeng
 * time: 2018-09-10 22:59
 */
public class SaveModeTest {
    public static void main(String[] args) {
        SparkConf conf = new SparkConf()
                .setAppName("SaveModeTest")
                .setMaster("local");

        JavaSparkContext sc = new JavaSparkContext(conf);
        SQLContext sqlContext = new SQLContext(sc);

        DataFrame peopleDf = sqlContext.read().format("json").load("hdfs://spark1:9000/spark-sql/source/people.json");
        peopleDf.save("hdfs://spark1:9000/spark-sql/output/java/people_savemode_test", "json", SaveMode.Append);

    }
}

File: sparkJava/src/main/java/com/beifeng/scala/sql/SaveModeTest.scala
package com.beifeng.scala.sql

import org.apache.spark.sql.{SQLContext, SaveMode}
import org.apache.spark.{SparkConf, SparkContext}

object SaveModeTest {

  def main(args: Array[String]): Unit = {
    val conf = new SparkConf()
      .setAppName("SaveModeTest")
      .setMaster("local")
    val sc = new SparkContext(conf)
    val sqlContext = new SQLContext(sc)

    val df = sqlContext.read.format("json").load("hdfs://spark1:9000/spark-sql/source/people.json")

    df.show()
    df.printSchema()

    df.save("hdfs://spark1:9000/spark-sql/output/scala/people_savemode_test", "json", SaveMode.Append)


  }
}
